kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/spec/ServerTemplate.java

```diff
@@ -241,4 +244,48 @@ public int hashCode() {
         result = 31 * result + (capabilities != null ? capabilities.hashCode() : 0);
         return result;
     }
+
+    public ServerTemplate cloneServerTemplate() {
+        synchronized (lock) {
+            ServerTemplate serverTemplate = new ServerTemplate();
+            serverTemplate.setId(this.getId());
+            serverTemplate.setName(this.getName());
+            containersSpec.forEach(c -> serverTemplate.addContainerSpec(new ContainerSpec(c.getId(), c.getContainerName(), c.getServerTemplateKey(), c.getReleasedId(), c.getStatus(), cloneContainerConfigMap(c.getConfigs()))));
+
+            final Map<Capability, ServerConfig> cloneServerConfig = new EnumMap<>(Capability.class);
+            cloneServerConfig.putAll(configs);
+            serverTemplate.setConfigs(cloneServerConfig);
+
+            serverInstances.forEach(serverInstance -> serverTemplate.addServerInstance(new ServerInstanceKey(serverInstance.getServerTemplateId(), serverInstance.getServerName(), serverInstance.getServerInstanceId(), serverInstance.getUrl())));
```

Comment:
https://github.com/kiegroup/droolsjbpm-integration/blob/4fa524a2fc11228738fbd62c8ad4d7f848178f3d/kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/spec/ServerTemplate.java#L104
In here, it will return new instance,so we can use getServerInstanceKeys()

Reply:
But there is no setter for the whole collection. Only constructor where we can set serverInstances as a whole.